quickwit-core/src/lib.rs

```diff
@@ -33,6 +33,41 @@ pub use index::{
     clear_cache_directory, get_cache_directory_path, remove_indexing_directory, IndexService,
     IndexServiceError,
 };
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, Debug)]
+pub struct QuickwitBuildInfo {
+    commit_version_tag: String,
+    cargo_pkg_version: String,
+    cargo_build_target: String,
+    commit_short_hash: String,
+    commit_date: String,
+    pub version: String,
+}
+
+impl QuickwitBuildInfo {
```

Comment:
That's one of the rare thing we don't care to test in unit test.
I'd just make it const.

```
pub struct QuickwitBuildInfo {
    commit_version_tag: &'static str,
    cargo_pkg_version: &'static str,
    cargo_build_target: &'static str,
    commit_short_hash: &'static str,
    commit_date: &'static str,
    pub version: &'static str,
}

const QUICKWIT_BUILD_INFO: QuickwitBuildInfo = build_quickwit_build_info()

const fn build_quickwit_build_info() {
    // ...
}
```

The main benefit is that you can get rid of the extra argument in quickwit-serve.
A second benefit is that the code express/enforce what it is. A bunch of information specific to the build.

Reply:
good point. I moved the build function in quickwit-serve.